src/integer_mod_q/mat_polynomial_ring_zq/from.rs

```diff
@@ -13,95 +13,118 @@
 //! The explicit functions contain the documentation.
 
 use super::MatPolynomialRingZq;
-use crate::{error::MathError, integer::MatPolyOverZ, integer_mod_q::ModulusPolynomialRingZq};
-use std::fmt::Display;
+use crate::{integer::MatPolyOverZ, integer_mod_q::ModulusPolynomialRingZq};
 
-impl MatPolynomialRingZq {
-    /// Creates a new matrix with `num_rows` rows, `num_cols` columns,
-    /// zeros as entries and `modulus` as the modulus.
+impl From<(&MatPolyOverZ, &ModulusPolynomialRingZq)> for MatPolynomialRingZq {
+    /// Create a new polynomial ring matrix of type [`MatPolynomialRingZq`].
     ///
     /// Parameters:
-    /// - `num_rows`: number of rows the new matrix should have
-    /// - `num_cols`: number of columns the new matrix should have
-    /// - `modulus`: the common modulus of the matrix entries
+    /// - `value`: is a tuple of `(matrix, modulus)`
+    ///     - `matrix`: defines the polynomial matrix
+    ///     - `modulus`: the modulus which defines the ring
```

Comment:
use other documentation style for tuple, see Q::from for a reference. If you don't mind, please also fix it in PolynomialRingZq